components/service/pocket/src/main/java/mozilla/components/service/pocket/api/PocketJSONParser.kt

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.pocket.api
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.service.pocket.logger
+import mozilla.components.support.ktx.android.org.json.mapNotNull
+import mozilla.components.support.ktx.android.org.json.tryGetInt
+import mozilla.components.support.ktx.android.org.json.tryGetString
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+private const val JSON_STORY_TITLE_KEY = "title"
+private const val JSON_STORY_URL_KEY = "url"
+private const val JSON_STORY_IMAGE_URL_KEY = "imageUrl"
+private const val JSON_STORY_PUBLISHER_KEY = "publisher"
+private const val JSON_STORY_CATEGORY_KEY = "category"
+private const val JSON_STORY_TIME_TO_READ_KEY = "timeToRead"
+
+/**
+ * Holds functions that parse the JSON returned by the Pocket API and converts them to more usable Kotlin types.
+ */
+internal class PocketJSONParser internal constructor() {
+    /**
+     * @return The stories, removing entries that are invalid, or null on error; the list will never be empty.
+     */
+    fun jsonToPocketApiStories(json: String): List<PocketApiStory>? = try {
+        val rawJSON = JSONObject(json)
+        val storiesJSON = rawJSON.getJSONArray(KEY_ARRAY_ITEMS)
+        val stories = storiesJSON.mapNotNull(JSONArray::getJSONObject) { jsonToPocketApiStory(it) }
+
+        // We return null, rather than the empty list, because devs might forget to check an empty list.
+        if (stories.isNotEmpty()) stories else null
+    } catch (e: JSONException) {
+        logger.warn("invalid JSON from the Pocket endpoint", e)
+        null
+    }
```

Comment:
question: We can assume that this JSON is always small enough so that we can parse it completely in memory, no need to have a streaming parser, right?

Reply:
There should at max be 450 stories.
But if a streaming parser would overall be better I'll look into that.

Reply:
There would only be 450 stories at max. Currently there are <100.
With a streaming parser we are hardcoding the properties order and would be a bit harder to set our default values when the endpoint sends incomplete data (this is a scenario happening and for which we must be prepared).
So if it's okay with you I'd prefer keeping the currently proposed implementation.

Reply:
Talked with Sebastian "offline". He suggested we can go with the current approach and watch Sentry for any reports regarding related issues.
Will also confirm with the Pocket team the payload size.